packages/@ourworldindata/grapher/src/tooltip/TooltipContents.tsx

```diff
@@ -241,24 +270,26 @@ export class TooltipTable extends React.Component<TooltipTableProps> {
                                         </span>
                                     )}
                                 </td>
-                                {zip(columns, values).map(([column, value]) => {
-                                    const missing = value === undefined
-                                    return column ? (
-                                        <td
-                                            key={column.slug}
-                                            className={classnames(
-                                                "series-value",
-                                                { missing }
-                                            )}
-                                        >
-                                            {!missing &&
-                                                column.formatValueShort(
-                                                    value,
-                                                    format
+                                {zip(columns, excludeUndefined(values)).map(
```

Comment:
I'm not sure about this change here: could this lead to a case where we're matching a value to the wrong col?

E.g. if `cols = [col1, col2, col3]` and `values = [1, undefined, 2]`, then we would end up with `col1: 1, col2: 2`, I think.

I'm also not sure why you changed this in the first place?

Reply:
I changed it because it gave a mobx warning (but I'm not sure which of my code changes made the mob warning appear).

Yeah, I thought about that. Although it's technically possible that we start matching the wrong columns/values, in practice, that doesn't happen if you look at how this is used. Still, it's not bulletproof, of course — maybe it's not worth the risk of getting rid of a warning.

Reply:
Hm, interesting. I kind-of-assume that the Mobx warning would go away if you use whatever way of copying that array, and it doesn't have to be `excludeUndefined` in particular.

Can you try `[...values]` instead and see if that works?

Reply:
`[...values]` didn't work. I think MobX's issue is that the zipped arrays don't have the same length, which means elements out of bounds are accessed. I got rid of the `excludeUndefined` now and implemented the opposite thing – if one array is shorter than the other, we fill it up with undefined values. It's a bit cumbersome, but it does get rid of the warning.

Reply:
Oh, that's confusing as hell!
Do you have a good sense of why `columns` and `values` can be of different length? Is there a good reason for that?
Otherwise, it would be good to get rid of that footgun tbh.

Reply:
Line charts pass [y-value, col-value or undefined] and [y-value column]. That's why I used excludeUndefined. Instead, I could use excludeUndefined in the line chart component (it should be safe since line charts without a y-col but with a color-col don't exist)